larrecodnn/ImagePatternAlgs/Modules/NoiseWaveformDump_module.cc

```diff
@@ -0,0 +1,324 @@
+    return;
+  }
+
+  // ... Build a map from channel number -> rawdigitVec
+  std::map<raw::ChannelID_t, art::Ptr<raw::RawDigit>> rawdigitMap;
+  raw::ChannelID_t chnum = raw::InvalidChannelID; // channel number
+  for (size_t rdIter = 0; rdIter < digitVecHandle->size(); ++rdIter) {
+    art::Ptr<raw::RawDigit> digitVec(digitVecHandle, rdIter);
+    chnum = digitVec->Channel();
+    if (chnum == raw::InvalidChannelID) continue;
+    if (geo::PlaneGeo::ViewName(fgeom->View(chnum)) != fPlaneToDump[0]) continue;
+    rawdigitMap[chnum] = digitVec;
+  }
+
+  // ... Read in sim channel list
+  auto simChannelHandle = evt.getValidHandle<std::vector<sim::SimChannel>>(fSimChannelLabel);
+
+  std::cout << " !!!!! Size of simChannelHandle: " << simChannelHandle->size() << std::endl;
+  if (!simChannelHandle->size()) return;
+
+  std::string dummystr6 = "none  ";
+  std::string dummystr7 = "none   ";
+
+  // ... Create a vector to hold channels numbers of signal and noise channels
+  std::vector<raw::ChannelID_t> signalchannels;
+  std::vector<raw::ChannelID_t> noisechannels;
+
+  // ... Loop over simChannels to find all signal channels and erase them
+  //     from the rawdigitMap
+  for (auto const& channel : (*simChannelHandle)) {
+
+    // .. get simChannel channel number
+    const raw::ChannelID_t ch1 = channel.Channel();
+    if (ch1 == raw::InvalidChannelID) continue;
+    if (geo::PlaneGeo::ViewName(fgeom->View(ch1)) != fPlaneToDump[0]) continue;
+
+    bool hasEnergyDeposit = false;
+
+    // ... Loop over all ticks with ionization energy deposited
+    auto const& timeSlices = channel.TDCIDEMap();
+    for (auto const& timeSlice : timeSlices) {
+
+      auto const& energyDeposits = timeSlice.second;
+      auto const tpctime = timeSlice.first;
+      unsigned int tdctick = static_cast<unsigned int>(clockData.TPCTDC2Tick(double(tpctime)));
+      if (tdctick < 0 || tdctick > (dataSize - 1)) continue;
+
+      // ... Loop over all energy depositions in this tick
+      for (auto const& energyDeposit : energyDeposits) {
+        if ((energyDeposit.energy > 0) || (energyDeposit.numElectrons > 0)) {
+          hasEnergyDeposit = true;
+          break;
+        }
+      }
+      if (hasEnergyDeposit) break;
+    }
+    if (hasEnergyDeposit) rawdigitMap.erase(ch1);
+
+  } // loop over SimChannels
+
+  // .. Now construct noise channel vector from updated rawdigitMap
+  for (std::map<raw::ChannelID_t, art::Ptr<raw::RawDigit>>::iterator iter = rawdigitMap.begin();
+       iter != rawdigitMap.end();
+       ++iter) {
+    noisechannels.push_back(iter->first);
+  }
+
+  std::cout << " !!!!! size of noisechannels: " << noisechannels.size() << std::endl;
+  if (!noisechannels.size()) return;
+
+  //save noise
+  int noisechancount = 0;
+
+  // .. create a vector for shuffling the wire channel indices
+  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Comment:
I guess you don't require reproducibility for this...?

Reply:
no I don't